lib/mozart_fetcher/fetcher.ex

```diff
@@ -12,14 +12,21 @@ defmodule MozartFetcher.Fetcher do
   def process(components) do
     ExMetrics.timeframe "function.timing.fetcher.process" do
       components
-      |> Enum.with_index
+      |> Enum.with_index()
       |> Enum.map(&Task.async(fn -> Component.fetch(&1) end))
-      |> Enum.map(&Task.await/1)
+      |> Enum.map(&Task.await(&1, max_timeout(components)))
       |> decorate_response
       |> Jason.encode!()
     end
   end
 
+  defp max_timeout(components) do
+    components
+    |> Enum.map(&Map.get(&1, :endpoint))
```

Comment:
instead of 2 maps, how about this reduce?
`Enum.reduce(components, 0, &Enum.max([TimeoutParser.parse(&1.endpoint), &2]))`
we can expand it to make it more readable...

Reply:
is it really faster with 50 items?

Reply:
we should *benchmark it* ! :)

Reply:
yes we should, but surely it is? Instead of iterating through the array 150 times, we do 52 times?

Reply:
our array shouldn't be that big, and sometimes the cost of instantiating new intermediate object can be counterproductive. This is not to say we shouldn't try, instead you're probably right! Just suggesting to be data driven

Reply:
Didn't the data show us an issue!